### PR TITLE
New signup snippet with test variants.

### PR DESCRIPTION
**Pull request by ScottDowne**
Hey @glogiotatidis 

Jean was asking if I had free time to do a new signup snippet.

Not sure yet if she has talked to you yet, so I might be ahead of the game, if so, I wouldn't worry about this until she talks to you. Otherwise, here it is :) 

Comps are here: https://mozilla.invisionapp.com/share/PVB3O3696#/screens/226586006

It's basically a new snippet, but I did copy and repurpose a lot of existing chunks of css/js from the other signup snippet and the simple snippet. It is actually quite a bit more simple than the last. There is also an added option to redirect to a url after successful signup, or use a simple success message. It also has three test variants, hover, static, and click. I think that's mostly it.

Thoughts?

### CODE REVIEW

File: generic-templates/newsletter-signup-snippet/snippet.html [hunk 1]
```diff
@@ -172,327 +198,214 @@
...
-          if (elHighlighted) {
-            elHighlighted.classList.add("active");
+
+          function changeState(status) {
+            if (status === 200) {
+              button.classList.remove("loading");
+              basketError.classList.add("hidden");
+              successMessage.classList.remove("hidden");
+              formContainer.classList.add("hidden");
+              sendMetric("signup-success");
```
Comment (glogiotatidis) [inline generic-templates/newsletter-signup-snippet/snippet.html]: Do you want to add to blocklist as well to not show the snippet to the user again?

File: generic-templates/newsletter-signup-snippet/snippet.html [hunk 2]
```diff
@@ -172,327 +198,214 @@
...
+               var xbutton = xbuttons[k];
+               xbutton.addEventListener('mouseover', mouseover, false);
+               xbutton.addEventListener('mouseout', mouseout, false);
+            }
+          {% endif %}
+
+          var elHighlighted = snippet.querySelector("em");
+          // Fade in highlight on text
+          setTimeout(function() {
+            if (elHighlighted) {
```
Comment (glogiotatidis) [inline generic-templates/newsletter-signup-snippet/snippet.html]: `if` can be outside of `setTimeout`

File: generic-templates/newsletter-signup-snippet/snippet.html [hunk 3]
```diff
@@ -172,327 +198,214 @@
...
+              basketError.classList.add("hidden");
+              successMessage.classList.remove("hidden");
+              formContainer.classList.add("hidden");
+              sendMetric("signup-success");
+            } else if (status === 000) {
+              //loading
+              button.classList.add("loading");
+              basketError.classList.add("hidden");
+              successMessage.classList.add("hidden");
+              {% if success_redirect %}
```
Comment (glogiotatidis) [inline generic-templates/newsletter-signup-snippet/snippet.html]: This should be in `status===200` right?

Comment (ScottDowne) [inline generic-templates/newsletter-signup-snippet/snippet.html]: Wow, yeah, goood catch.

**Comment by ScottDowne** [comment]
@glogiotatidis Fixed up! :)

File: generic-templates/newsletter-signup-snippet/snippet.html [hunk 4]
```diff
@@ -172,327 +199,217 @@
...
-          transitionTrack.style.transform = "translate(-" + ( currentIndex * trackItemWidth ) + "px, 0px)";
-        }
-        function onBack(e) {
-          if (currentIndex <= 0) {
-            return;
+
+          function changeState(status) {
+            if (status === 200) {
+              sendMetric("signup-success", function() {
+                addToBlockList();
```
Comment (glogiotatidis) [inline generic-templates/newsletter-signup-snippet/snippet.html]: I'll add a delay here in the case of redirect.

**Comment by glogiotatidis** [comment]
Thanks!

**Pull request merged**